Wikipedia/Code/ReadingListEntryCollectionViewController.swift

```diff
@@ -230,6 +218,31 @@ class ArticlesCollectionViewController: ColumnarCollectionViewController, Editab
         return ColumnarCollectionViewLayoutMetrics.tableViewMetrics(with: size, readableWidth: readableWidth, layoutMargins: layoutMargins)
     }
     
+    // MARK: - Article changes
+    
+    @objc func articleDidChange(_ note: Notification) {
+        guard
+            let article = note.object as? WMFArticle,
+            article.hasChangedValuesForCurrentEventThatAffectSavedArticlePreviews,
+            let articleKey = article.key
+            else {
+                return
+        }
+        
+        let visibleIndexPathsWithChanges = collectionView.indexPathsForVisibleItems.filter { (indexPath) -> Bool in
+            guard let entry = entry(at: indexPath) else {
+                return false
+            }
+            return entry.articleKey == articleKey
+        }
+        
+        guard visibleIndexPathsWithChanges.count > 0 else {
```

Comment:
According to https://developer.apple.com/documentation/swift/array/1688398-isempty, it'd be better to check `isEmpty` than accessing `count` in this case -

> When you need to check whether your collection is empty, use the isEmpty property instead of checking that the count property is equal to zero. For collections that don’t conform to RandomAccessCollection, accessing the count property iterates through the elements of the collection.

Reply:
In this case, an array of index paths is a `RandomAccessCollection` so it's still constant time, but it's good to get into the habit